modules/dicom_archive/php/dicom_archive.class.inc

```diff
@@ -256,14 +256,14 @@ class Dicom_Archive extends \NDB_Menu_Filter
 
         foreach ($unanonymized['Data'] as &$row) {
             $val = $row[2];
-            if (!preg_match($this->dicomArchiveSettings['patientNameRegex'], $val)
-                && !preg_match($this->dicomArchiveSettings['LegoPhantomRegex'], $val)
-                && !preg_match(
+            if (! preg_match($this->dicomArchiveSettings['patientNameRegex'], $val)
+                or !preg_match($this->dicomArchiveSettings['LegoPhantomRegex'], $val)
+                or !preg_match(
```

Comment:
I don't think this change is right. A scan is either labeled as a patient or as a lego, which means the "!preg_match" should be an and in order for it to work. (ie. if it's not either a valid patient label OR a lego label it's invalid, but the keyword needs to be '&&' since the condition is negated..)

(Also, why is this using `or` and not `||`?)

Reply:
I see. Then it should be made clear in the frontend that regex for **patient name, lego phantom AND living phantom** need to be set for the invalid to work. Studies that don't have lego or living phantom datasets won't update the regex for these entries, which by default are /./i..., hence it becomes useless to set the patient name regex.

@driusan should we change the default value for the regex of phantom datasets to avoid confusion or should we document this somehow in the config module on the frontend (which might be quite dirty) Let me know. 

I'll close this PR since the DICOM archive module is doing what is expected already

Thanks!!

Reply:
@driusan mmmm, technically, couldn't there be only one regex instead of three?
/PSCID_\d\d\d\d\d\d_V\d | lego_phantom | living_phantom/i
This would be a lot less confusing than remembering the logic of the module, which won't apply to studies without phantom datasets.

Just an idea...

Reply:
That's a good point about it not working correctly if only one is set, but I think putting out it all into one regex would make configuration a pain. Changing the defaults seems like a reasonable solution to me, though.

Reply:
@driusan I issued a new PR to change the default configuration for lego and living phantom regex. Thanks! https://github.com/aces/Loris/pull/3443